### PR TITLE
Restart environment

### LINKED ISSUE #180: Restart environment
A user should be able to restart an environment. It will:
- stop exiting containers
- rebuild the image
- start the containers with updated images

### PR DESCRIPTION
**Pull request by feloy**
### What does this PR do?

Add a button to restart an environment from the Environments page, using the new source code

### Screenshot / video of UI

https://github.com/projectatomic/ai-studio/assets/9973512/5ab2f616-974a-4bc9-b7e6-aa1d24717271

### What issues does this PR fix or reference?

Fixes #180 

### How to test this PR?

- start an app from the recipe page
- go to Environments page
- click Restart button, the pod should be deleted and recreated
- modify sources of application and/ model service
- click Restart button, the pod should be deleted and recreated, using the new source code

### CODE REVIEW

File: packages/frontend/src/lib/table/environment/ColumnActions.svelte [hunk 1]
```diff
@@ -11,11 +12,27 @@ function deleteEnvironment() {
...
 </script>
 
-<ListItemButtonIcon
-  icon={faTrash}
-  onClick={() => deleteEnvironment()}
-  title="Delete Environment"
-  inProgress={object.status === 'stopping' || object.status === 'removing'}
-/>
+
+{#if object.status === 'stopping' || object.status === 'removing'}
```
Comment (axel7083) [inline packages/frontend/src/lib/table/environment/ColumnActions.svelte:23]: I like the upgrade!

**Review by jeffmaury — COMMENTED** [review]
I got the env running where in fact the model service container has not yet fully loaded the model and thus the client application has crashed

**Comment by feloy** [comment]
> I got the env running where in fact the model service container has not yet fully loaded the model and thus the client application has crashed

I cannot see where the problem occurs. What are the operations you have done in the UI?

**Comment by jeffmaury** [comment]
> > I got the env running where in fact the model service container has not yet fully loaded the model and thus the client application has crashed
> 
> I cannot see where the problem occurs. What are the operations you have done in the UI?

Just started an env then switch to the env tabs and restarted it

**Comment by feloy** [comment]
> > > I got the env running where in fact the model service container has not yet fully loaded the model and thus the client application has crashed
> > 
> > 
> > I cannot see where the problem occurs. What are the operations you have done in the UI?
> 
> Just started an env then switch to the env tabs and restarted it

Is it reproducible? On my side, I cannot see this happen

**Comment by jeffmaury** [comment]
> > > > I got the env running where in fact the model service container has not yet fully loaded the model and thus the client application has crashed
> > > 
> > > 
> > > I cannot see where the problem occurs. What are the operations you have done in the UI?
> > 
> > 
> > Just started an env then switch to the env tabs and restarted it
> 
> Is it reproducible? On my side, I cannot see this happen

restart

**Comment by feloy** [comment]
> > > > > I got the env running where in fact the model service container has not yet fully loaded the model and thus the client application has crashed
> > > > 
> > > > 
> > > > I cannot see where the problem occurs. What are the operations you have done in the UI?
> > > 
> > > 
> > > Just started an env then switch to the env tabs and restarted it
> > 
> > 
> > Is it reproducible? On my side, I cannot see this happen
> 
> restart [ restart ](https://private-user-images.githubusercontent.com/695993/304430744-d37176e1-5c95-49dc-bf75-5e3c6e485bb6.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MzI0NDYsIm5iZiI6MTcwNzgzMjE0NiwicGF0aCI6Ii82OTU5OTMvMzA0NDMwNzQ0LWQzNzE3NmUxLTVjOTUtNDlkYy1iZjc1LTVlM2M2ZTQ4NWJiNi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QxMzQ5MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTM0MDRlNTJlMDE1YjI0YTk1YTM1ZWQwOTVmNjExYzA0ZGUwODU5OTI1ZWZiZmUwM2IwY2U4ZjI2MjBmNzU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MBn6_OlQ4ltKvzQdcN2UFcyG4-AqtVMGclx5lFhxtmk) [ ](https://private-user-images.githubusercontent.com/695993/304430744-d37176e1-5c95-49dc-bf75-5e3c6e485bb6.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MzI0NDYsIm5iZiI6MTcwNzgzMjE0NiwicGF0aCI6Ii82OTU5OTMvMzA0NDMwNzQ0LWQzNzE3NmUxLTVjOTUtNDlkYy1iZjc1LTVlM2M2ZTQ4NWJiNi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QxMzQ5MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTM0MDRlNTJlMDE1YjI0YTk1YTM1ZWQwOTVmNjExYzA0ZGUwODU5OTI1ZWZiZmUwM2IwY2U4ZjI2MjBmNzU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MBn6_OlQ4ltKvzQdcN2UFcyG4-AqtVMGclx5lFhxtmk)

The status displayed in Environment page and Recipe page are misleading, but if you would have wait for the model to restart (after pressing the Restart button), your app should have restarted correctly.

I have following PRs that will help fix the display of the status (but at the end, this will be a big refactoring, as discussed this morning, so mant PRs are coming), but I would prefer to keep this one small.

**Comment by jeffmaury** [comment]
> > > > > > I got the env running where in fact the model service container has not yet fully loaded the model and thus the client application has crashed
> > > > > 
> > > > > 
> > > > > I cannot see where the problem occurs. What are the operations you have done in the UI?
> > > > 
> > > > 
> > > > Just started an env then switch to the env tabs and restarted it
> > > 
> > > 
> > > Is it reproducible? On my side, I cannot see this happen
> > 
> > 
> > restart
> >       
> >         [
> >           
> >         
> >             restart
> >           ](https://private-user-images.githubusercontent.com/695993/304430744-d37176e1-5c95-49dc-bf75-5e3c6e485bb6.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MzI0NDYsIm5iZiI6MTcwNzgzMjE0NiwicGF0aCI6Ii82OTU5OTMvMzA0NDMwNzQ0LWQzNzE3NmUxLTVjOTUtNDlkYy1iZjc1LTVlM2M2ZTQ4NWJiNi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QxMzQ5MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTM0MDRlNTJlMDE1YjI0YTk1YTM1ZWQwOTVmNjExYzA0ZGUwODU5OTI1ZWZiZmUwM2IwY2U4ZjI2MjBmNzU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MBn6_OlQ4ltKvzQdcN2UFcyG4-AqtVMGclx5lFhxtmk)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/695993/304430744-d37176e1-5c95-49dc-bf75-5e3c6e485bb6.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MzI0NDYsIm5iZiI6MTcwNzgzMjE0NiwicGF0aCI6Ii82OTU5OTMvMzA0NDMwNzQ0LWQzNzE3NmUxLTVjOTUtNDlkYy1iZjc1LTVlM2M2ZTQ4NWJiNi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QxMzQ5MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTM0MDRlNTJlMDE1YjI0YTk1YTM1ZWQwOTVmNjExYzA0ZGUwODU5OTI1ZWZiZmUwM2IwY2U4ZjI2MjBmNzU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MBn6_OlQ4ltKvzQdcN2UFcyG4-AqtVMGclx5lFhxtmk)
> >         
> >        [ restart ](https://private-user-images.githubusercontent.com/695993/304430744-d37176e1-5c95-49dc-bf75-5e3c6e485bb6.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MzI0NDYsIm5iZiI6MTcwNzgzMjE0NiwicGF0aCI6Ii82OTU5OTMvMzA0NDMwNzQ0LWQzNzE3NmUxLTVjOTUtNDlkYy1iZjc1LTVlM2M2ZTQ4NWJiNi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QxMzQ5MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTM0MDRlNTJlMDE1YjI0YTk1YTM1ZWQwOTVmNjExYzA0ZGUwODU5OTI1ZWZiZmUwM2IwY2U4ZjI2MjBmNzU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MBn6_OlQ4ltKvzQdcN2UFcyG4-AqtVMGclx5lFhxtmk)
> >       
> >         [
> >           
> >         
> >             restart
> >           ](https://private-user-images.githubusercontent.com/695993/304430744-d37176e1-5c95-49dc-bf75-5e3c6e485bb6.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MzI0NDYsIm5iZiI6MTcwNzgzMjE0NiwicGF0aCI6Ii82OTU5OTMvMzA0NDMwNzQ0LWQzNzE3NmUxLTVjOTUtNDlkYy1iZjc1LTVlM2M2ZTQ4NWJiNi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QxMzQ5MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTM0MDRlNTJlMDE1YjI0YTk1YTM1ZWQwOTVmNjExYzA0ZGUwODU5OTI1ZWZiZmUwM2IwY2U4ZjI2MjBmNzU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MBn6_OlQ4ltKvzQdcN2UFcyG4-AqtVMGclx5lFhxtmk)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://private-user-images.githubusercontent.com/695993/304430744-d37176e1-5c95-49dc-bf75-5e3c6e485bb6.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MzI0NDYsIm5iZiI6MTcwNzgzMjE0NiwicGF0aCI6Ii82OTU5OTMvMzA0NDMwNzQ0LWQzNzE3NmUxLTVjOTUtNDlkYy1iZjc1LTVlM2M2ZTQ4NWJiNi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QxMzQ5MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTM0MDRlNTJlMDE1YjI0YTk1YTM1ZWQwOTVmNjExYzA0ZGUwODU5OTI1ZWZiZmUwM2IwY2U4ZjI2MjBmNzU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MBn6_OlQ4ltKvzQdcN2UFcyG4-AqtVMGclx5lFhxtmk)
> >         
> >        [ ](https://private-user-images.githubusercontent.com/695993/304430744-d37176e1-5c95-49dc-bf75-5e3c6e485bb6.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4MzI0NDYsIm5iZiI6MTcwNzgzMjE0NiwicGF0aCI6Ii82OTU5OTMvMzA0NDMwNzQ0LWQzNzE3NmUxLTVjOTUtNDlkYy1iZjc1LTVlM2M2ZTQ4NWJiNi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QxMzQ5MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTM0MDRlNTJlMDE1YjI0YTk1YTM1ZWQwOTVmNjExYzA0ZGUwODU5OTI1ZWZiZmUwM2IwY2U4ZjI2MjBmNzU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MBn6_OlQ4ltKvzQdcN2UFcyG4-AqtVMGclx5lFhxtmk)
> 
> The status displayed in Environment page and Recipe page are misleading, but if you would have wait for the model to restart (after pressing the Restart button), your app should have restarted correctly.
> 
> I have following PRs that will help fix the display of the status (but at the end, this will be a big refactoring, as discussed this morning, so mant PRs are coming), but I would prefer to keep this one small.

Yes I then discovered that the status in environment page is wrong but not related to this PR so LGTM

**Review by jeffmaury — APPROVED** [review]

**Pull request merged**